### PR TITLE
ENH: Type hints in functional interface

### PR DESCRIPTION
**Pull request by tbirdso**
Added type hints to Python functional interface methods in `extras.py`. This will close [Issue 2446](https://github.com/InsightSoftwareConsortium/ITK/issues/2446).

## PR Checklist
- [x] No [API changes](https://github.com/InsightSoftwareConsortium/ITK/blob/master/CONTRIBUTING.md#breaking-changes) were made
- [x] No [major design changes](https://github.com/InsightSoftwareConsortium/ITK/blob/master/CONTRIBUTING.md#design-changes) were made

### CODE REVIEW

**Review by thewtex — CHANGES_REQUESTED** [review]
Thanks @tbirdso !

A few items noted inline.

Also, please use `np.ndarray` instead of `numpy.ndarray` throughout to follow Python conventions.

File: Wrapping/Generators/Python/itk/support/extras.py [hunk 1]
```diff
@@ -535,30 +563,34 @@ def _GetVnlObjectFromArray(arr, function_name: str, ttype):
...
-def GetArrayFromMatrix(itk_matrix):
+def GetArrayFromMatrix(itk_matrix) -> numpy.ndarray:
     return GetArrayFromVnlMatrix(itk_matrix.GetVnlMatrix().as_matrix())
 
 
 array_from_matrix = GetArrayFromMatrix
 
 
-def GetMatrixFromArray(arr):
+def GetMatrixFromArray(arr : npt.ArrayLike):
```
[the remark this answers is not in the document]
Comment (tbirdso) [reply Wrapping/Generators/Python/itk/support/extras.py]: Do we also want to add `VnlObject` type hints for methods such as `_GetVnlObjectFromArray`?

**Review by dzenanz — COMMENTED** [review]
LGTM

File: Wrapping/Generators/Python/itk/support/extras.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (thewtex) [reply Wrapping/Generators/Python/itk/support/extras.py]: We eventually want to get rid of VNL, so let's not add that.

**Review by thewtex — APPROVED** [review]

**Review by hjmjohnson — APPROVED** [review]

**Pull request merged**